### PR TITLE
refactor(localize): miscellaneous refactorings

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: packages/localize/src/tools/src/source_file_utils.ts [hunk 1]
```diff
@@ -355,3 +354,12 @@ export function buildCodeFrameError(path: NodePath, e: BabelParseError): string
...
   return `${filename}: ${message}`;
 }
+
+export function getLocation(path: NodePath): ɵSourceLocation|undefined {
+  return path.node.loc && path.hub.file.opts.filename ? {
+    start: {...path.node.loc.start},
+    end: {...path.node.loc.end},
+    file: path.hub.file.opts.filename
+  } :
+                                                        undefined;
```
Comment (AndrewKushnir) [inline packages/localize/src/tools/src/source_file_utils.ts:364]: Formatting is a bit unusual :) Maybe do an `if` instead?

File: packages/localize/src/utils/src/messages.ts [hunk 2]
```diff
@@ -38,6 +38,52 @@ export type TargetMessage = string;
...
+   */
+  legacyIds?: string[];
+  /**
+   * The meaning of the `message`, used to distinguish identical `messageString`s.
+   */
+  meaning?: string;
+  /**
+   * The description of the `message`, used to aid translation.
+   */
+  description?: string|undefined;
```
Comment (AndrewKushnir) [inline packages/localize/src/utils/src/messages.ts:80]: I guess we don't need `undefined` here?

**Review by AndrewKushnir — APPROVED** [review]
LGTM, thanks @petebacondarwin 👍  Just left a couple minor comments.

**Comment by matsko** [comment]
@petebacondarwin this fails on patch. Please open a new PR to address the `patch` branch directly.

**Comment by matsko** [comment]
@petebacondarwin please rebase.

**Comment by petebacondarwin** [comment]
@matsko  - done.

**Comment by petebacondarwin** [comment]
@matsko - I assume that you removed the `patch` target label because this will not land cleanly on patch rather than because it is not valid for patch?

**Pull request closed without merging**